### ISSUE TITLE
Mock writeHead doesn't return response

### ISSUE THREAD

**Issue by herobank110** [howardabrams/node-mocks-http#228]
**Overview**
Mock writeHead doesn't return response so can't be chained as usual.

**Use Case**
Normally `ServerResponse.writeHead` returns the res object so it would be valid code to chain a function after it.
But `httpMocks.MockResponse<ServerResponse>` returns undefined causing valid code to fail.

**Version(s)**
node-mocks-http 1.9.0

**Reproduce** 
```typescript
// main app
const handler = (req: IncomingMessage, res: ServerResponse) => { res.writeHead(200).end(); };
http.createServer(handler);

// tests
const req = httpMocks.createRequest<IncomingMessage>();
const res = httpMocks.createResponse<ServerResponse>();
handler(req, res);
```

**Suggestions**
Workaround in typescript when making a mock response
```typescript
const res = httpMocks.createResponse<ServerResponse>();
const g = res.writeHead;
// @ts-ignore
res.writeHead = (..._) => g(..._) || res;
```

**Comment by eugef** [comment]
@herobank110 please check if this fix works for you #229

**Comment by herobank110** [comment]
Yes this change works fixes my issue with my workaround removed, Many Thanks!
